### PR TITLE
DirectionsProfile for reroutes in NavigationView

### PR DESCRIPTION
**Pull request by danesfeder**
- Add directions profile to navigation view options for reroute scenarios
- If no profile is provided, and the UI was launched with a `DirectionsRoute` the default for reroutes will be the profile from the route used to launch the UI 
- If no profile is provided, and the UI was launched with `Position` coordinates, the default will be `DirectionsCriteria.PROFILE_DRIVING_TRAFFIC`

cc @ericrwolfe

### CODE REVIEW

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java [hunk 1]
```diff
@@ -157,6 +159,8 @@ private boolean launchWithRoute(NavigationViewOptions options) {
   private void extractRoute(NavigationViewOptions options) {
     DirectionsRoute route = options.directionsRoute();
     if (route != null) {
+      String profile = options.directionsProfile();
+      routeProfile = profile != null ? profile : route.routeOptions().profile();
```
Comment (cammace) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java]: Several checks are already in place to ensure this is never null including the [MAS Directions builder](https://github.com/mapbox/mapbox-java/blob/master/services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java#L331) and the server response itself. I've opened https://github.com/mapbox/mapbox-java/issues/683 to change the annotation for this to a NonNull

Comment (danesfeder) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java]: @cammace awesome, thanks for the heads up on this

Comment (darkwizzard) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java]: Hey, it is posible to pass Locale when reroute? seems to be the answer for issue #633

Comment (danesfeder) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java]: @darkwizzard PR is waiting for review https://github.com/mapbox/mapbox-navigation-android/pull/635

**Pull request merged**